File: cam/CAM.DataTransferObjects/VIA/ViaExport.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using CAM.DataAttributes.Grid;
using CAM.DataTransferObjects.FunctionalityDto;
using CAM.DataTransferObjects.QueryDto.Base;
using CAM.Enum;

namespace CAM.DataTransferObjects.VIA
{
    public class ViaExport : IEquatable<ViaExport>
    {
        [OrderGrid(Order = 1)]
        [DisplayName("ME_SOURCE_ASSET_ID ")]
        [Default]
        public string MeSourceAssetId { get; set; }

        [DisplayName("ME_NAME")]
        [OrderGrid(Order = 2)]
        [Default]
        public string MeName { get; set; }

        [DisplayName("ME_TYPE")]
        [OrderGrid(Order = 3)]
        [Default]
        public string MeType { get; set; }

        [DisplayName("LCM_PROD_NAME")]
        [OrderGrid(Order = 5)]
        [Default]
        public string LcmProdName { get; set; }

        [DisplayName("VENDOR")]
        [OrderGrid(Order = 6)]
        [Default]
        public string Vendor { get; set; }

        [DisplayName("EOSL_CONTRACT_DATE")]
        [OrderGrid(Order = 7)]
        [DateRangeGridString]
        [Default]
        public string EoslContractDate { get; set; }

        [DisplayName("EEOSL_CONTRACT_DATE")]
        [OrderGrid(Order = 8)]
        [DateRangeGridString]
        [Default]
        public string EeoslContractDate { get; set; }

        [DisplayName("LOCATION_NAME")]
        [OrderGrid(Order = 9)]
        [Default]
        public string LocationName { get; set; }

        [DisplayName("ME_DESCRIPTION")]
        [OrderGrid(Order = 10)]
        [Default]
        public string MeDescription { get; set; }

        [DisplayName("ME_STATUS")]
        [OrderGrid(Order = 11)]
        [Default]
        public string MeStatus { get; set; }

        [DisplayName("ME_IP_ADDRESS")]
        [OrderGrid(Order = 12)]
        [Default]
        public string MeIpAddress { get; set; }

        [DisplayName("ME_VIRTUAL_FLG")]
        [OrderGrid(Order = 13)]
        [Default]
        public string MeVirtualFlg { get; set; }

        [DisplayName("ORGANISATION_NAME")]
        [OrderGrid(Order = 14)]
        [Default]
        public string OrganisationName { get; set; }

        [DisplayName("OS_NAME")]
        [OrderGrid(Order = 15)]
        [Default]
        public string OsName { get; set; }

        [DisplayName("OS_EOSL_CONTRACT_DATE")]
        [OrderGrid(Order = 16)]
        [DateRangeGridString]
        [Default]
        public string OsEoslContractDate { get; set; }

        [DisplayName("OS_EEOSL_CONTRACT_DATE")]
        [OrderGrid(Order = 17)]
        [DateRangeGridString]
        [Default]
        public string OsEeoslContractDate { get; set; }

        [DisplayName("LCMPolicy")]
        [OrderGrid(Order = 18)]
        [Default]
        public string MeLCMPolicy { get; set; }

        [DisplayName("Business Critical")]
        [OrderGrid(Order = 19)]
        [Default]
        public string MeCniBcServiceFlg { get; set; }

        [DisplayName("ME_CRITICALITY")]
        [OrderGrid(Order = 20)]
        [Default]
        public string MeCriticality { get; set; }

        [DisplayName("ME_DEPLOYMENT_TYPE")]
        [OrderGrid(Order = 21)]
        [Default]
        public string MeDeploymentType { get; set; }

        [DisplayName("ME_ENVIRONMENT")]
        [OrderGrid(Order = 22)]
        [Default]
        public string MeEnvironment { get; set; }

        [DisplayName("ME_EXTERNAL_CONNECTION_FLG")]
        [OrderGrid(Order = 23)]
        [Default]
        public string MeExternalConnectionFlg { get; set; }

        [DisplayName("ME_FQDN")]
        [OrderGrid(Order = 24)]
        [Default]
        public string MeFqdn { get; set; }

        [DisplayName("ME_GDPR_RELEVANT_FLG")]
        [OrderGrid(Order = 25)]
        [Default]
        public string MeGdprRelevantFlg { get; set; }

        [DisplayName("ME_INSTALLATION_DATE")]
        [OrderGrid(Order = 26)]
        [Default]
        //[DateRangeGridString]
        public string MeInstallationDate { get; set; }

        [DisplayName("ME_LAST_MAJOR_UPGRADE_DATE")]
        [OrderGrid(Order = 27)]
        [DateRangeGridString]
        [Default]
        public string MeLastMajorUpgradeDate { get; set; }

        [DisplayName("ME_LAST_SCAN_DATE")]
        [OrderGrid(Order = 28)]
        [DateRangeGridString]
        [Default]
        public string MeLastScanDate { get; set; }

        [DisplayName("ME_MAC_ADDRESS")]
        [OrderGrid(Order = 29)]
        [Default]
        public string MeMacAddress { get; set; }

        [DisplayName("ME_SERIAL_NUMBER")]
        [OrderGrid(Order = 30)]
        [Default]
        public string MeSerialNumber { get; set; }

        [DisplayName("ME_SERVICE_TYPE")]
        [OrderGrid(Order = 31)]
        [Default]
        public string MeServiceType { get; set; }

        [DisplayName("ME_CYBERARK_INTEGRATION_FLG")]
        [OrderGrid(Order = 32)]
        [Default]
        public string MeCyberarkIntegrationFlg { get; set; }

        [DisplayName("ME_IDM_INTEGRATION_FLG")]
        [OrderGrid(Order = 33)]
        [Default]
        public string MeIdmIntegrationFlg { get; set; }

        [DisplayName("ME_SECURITY_TIER")]
        [OrderGrid(Order = 34)]
        [Default]
        public string MeSecurityTier { get; set; }


        [DisplayName("ME_2FA_INTEGRATION_FLG")]
        [OrderGrid(Order = 35)]
        [Default]
        public string Me2FaIntegrationFlg { get; set; }

        [DisplayName("ME_SIEM_INTEGRATION_FLG")]
        [OrderGrid(Order = 36)]
        [Default]
        public string MeSiemIntegrationFlg { get; set; }

        [DisplayName("SUPPORT_CONTRACT")]
        [OrderGrid(Order = 37)]
        [Default]
        public string SupportContract { get; set; }

        [DisplayName("LCM_STATUS")]
        [OrderGrid(Order = 38)]
        [Default]
        public string LcmStatus { get; set; }

        [IgnoreGrid]
        [OrderGrid(Order = 39)]
        [Default]
        public EOMEnum EOMStatus { get; set; }
        public bool Equals(ViaExport other)
        {
            if (ReferenceEquals(null, other)) return false;
            if (ReferenceEquals(this, other)) return true;
            return MeSourceAssetId == other.MeSourceAssetId;
        }

        public override bool Equals(object obj)
        {
            if (ReferenceEquals(null, obj)) return false;
            if (ReferenceEquals(this, obj)) return true;
            if (obj.GetType() != this.GetType()) return false;
            return Equals((ViaExport)obj);
        }

        public override int GetHashCode()
        {
            var hashCode = new HashCode();
            hashCode.Add(MeSourceAssetId);
            return hashCode.ToHashCode();
        }
        [DisplayName("ResourceKey")]
        [OrderGrid(Order = 4)]
        [Default]
        public string resourceKey { get; set; }

        [DisplayName("Vertical Name")]
        [OrderGrid(Order = 40)]
        [Default]
        public string VerticalName { get; set; }

        
        [IgnoreGrid]
        public int VerticalNameId { get; set; }

    }

    public class ViaExportQuery : QueryObject
    {
        public List<string> MeSourceAssetId { get; set; }
        public List<string> MeName { get; set; }
        public List<string> MeType { get; set; }
        public List<string> LcmProdName { get; set; }
        public List<string> Vendor { get; set; }
        public DateFilter EoslContractDate { get; set; }
        public DateFilter EeoslContractDate { get; set; }
        public List<string> LocationName { get; set; }
        public List<string> MeDescription { get; set; }
        public List<string> MeStatus { get; set; }
        public List<string> MeIpAddress { get; set; }
        public List<string> MeVirtualFlg { get; set; }
        public List<string> OrganisationName { get; set; }

        public List<short> OrganisationNameId { get; set; }
        public List<string> OsName { get; set; }
        public DateFilter OsEoslContractDate { get; set; }
        public DateFilter OsEeoslContractDate { get; set; }
        public List<string> MeCniBcServiceFlg { get; set; }
        public List<string> MeCriticality { get; set; }
        public List<short> MeLCMPolicy { get; set; }
        public List<string> MeDeploymentType { get; set; }
        public List<string> MeEnvironment { get; set; }
        public List<string> MeExternalConnectionFlg { get; set; }
        public List<string> MeFqdn { get; set; }
        public List<string> MeGdprRelevantFlg { get; set; }
        public List<string> MeInstallationDate { get; set; }
        public DateFilter MeLastMajorUpgradeDate { get; set; }
        public DateFilter MeLastScanDate { get; set; }
        public List<string> MeMacAddress { get; set; }
        public List<string> MeSerialNumber { get; set; }
        public List<string> MeServiceType { get; set; }
        public List<string> MeCyberarkIntegrationFlg { get; set; }
        public List<string> MeIdmIntegrationFlg { get; set; }
        public List<string> MeSecurityTier { get; set; }
        public List<string> Me2FaIntegrationFlg { get; set; }
        public List<string> MeSiemIntegrationFlg { get; set; }
        public List<string> SupportContract { get; set; }
        public List<string> LcmStatus { get; set; }
        public List<string>ResourceKey { get; set; }
        public int SkipAmount { get; set; }
        public List<SkipManager> SkipManager { get; set; }

        public List<string> VerticalName { get; set; }

        public List<int> VerticalNameId { get; set; }
    }

    public class SkipManager
    {
        public int Page { get; set; }
        public int SkipAmount { get; set; }
    }
}

File: cam/CAM.WebAPI/ClientApp/src/Model/ViaExport/ViaExport.ts
/* eslint-disable @typescript-eslint/consistent-type-assertions */
import { FilterValueDto } from "../../Business/Common/CommonBusiness";
import { DateFilter, QueryObject, QueryObjectGrid, RenderDetail } from "../Common";
import { ResultDto } from "../CommonModels";


/**
 * 
 * @export
 * @interface SkipManager
 */
 export interface SkipManager {
    /**
     * 
     * @type {number}
     * @memberof SkipManager
     */
    page?: number;
    /**
     * 
     * @type {number}
     * @memberof SkipManager
     */
    skipAmount?: number;
}


/**
 * 
 * @export
 * @interface ViaExport
 */
 export interface ViaExport {
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meSourceAssetId?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meName?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meType?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    lcmProdName?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    vendor?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    eoslContractDate?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    eeoslContractDate?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    locationName?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meDescription?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meStatus?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meIpAddress?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meVirtualFlg?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    organisationName?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    osName?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    osEoslContractDate?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    osEeoslContractDate?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meCniBcServiceFlg?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meCriticality?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meDeploymentType?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meEnvironment?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meExternalConnectionFlg?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meFqdn?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meGdprRelevantFlg?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meInstallationDate?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meLastMajorUpgradeDate?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meLastScanDate?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meMacAddress?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meSerialNumber?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meServiceType?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meCyberarkIntegrationFlg?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meIdmIntegrationFlg?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meSecurityTier?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    me2FaIntegrationFlg?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    meSiemIntegrationFlg?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    supportContract?: string;
    /**
     * 
     * @type {string}
     * @memberof ViaExport
     */
    lcmStatus?: string;
}

/**
 * 
 * @export
 * @interface ViaExportQuery
 */
 export interface ViaExportQuery extends QueryObject {
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meSourceAssetId?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meName?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meType?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    lcmProdName?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    vendor?: Array<string>;
    /**
     * 
     * @type {DateFilter}
     * @memberof ViaExportQuery
     */
    eoslContractDate?: DateFilter;
    /**
     * 
     * @type {DateFilter}
     * @memberof ViaExportQuery
     */
    eeoslContractDate?: DateFilter;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    locationName?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meDescription?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meStatus?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meIpAddress?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meVirtualFlg?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    organisationName?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    osName?: Array<string>;
    /**
     * 
     * @type {DateFilter}
     * @memberof ViaExportQuery
     */
    osEoslContractDate?: DateFilter;
    /**
     * 
     * @type {DateFilter}
     * @memberof ViaExportQuery
     */
    osEeoslContractDate?: DateFilter;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meCniBcServiceFlg?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meCriticality?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meDeploymentType?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meEnvironment?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meExternalConnectionFlg?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meFqdn?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meGdprRelevantFlg?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meInstallationDate?: Array<string>;
    /**
     * 
     * @type {DateFilter}
     * @memberof ViaExportQuery
     */
    meLastMajorUpgradeDate?: DateFilter;
    /**
     * 
     * @type {DateFilter}
     * @memberof ViaExportQuery
     */
    meLastScanDate?: DateFilter;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meMacAddress?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meSerialNumber?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meServiceType?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meCyberarkIntegrationFlg?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meIdmIntegrationFlg?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meSecurityTier?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    me2FaIntegrationFlg?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    meSiemIntegrationFlg?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    supportContract?: Array<string>;
    /**
     * 
     * @type {Array<string>}
     * @memberof ViaExportQuery
     */
    lcmStatus?: Array<string>;
    /**
     * 
     * @type {number}
     * @memberof ViaExportQuery
     */
    skipAmount?: number;
    /**
     * 
     * @type {Array<SkipManager>}
     * @memberof ViaExportQuery
     */
    skipManager?: Array<SkipManager>;
}

/**
 * 
 * @export
 * @interface CustomGridRenderOfViaExport
 */
 export interface CustomGridRenderOfViaExport {
    /**
     * 
     * @type {string}
     * @memberof CustomGridRenderOfViaExport
     */
    className?: string;
    /**
     * 
     * @type {Array<RenderDetail>}
     * @memberof CustomGridRenderOfViaExport
     */
    render?: Array<RenderDetail>;
}

/**
 * 
 * @export
 * @interface QueryResultDtoOfViaExport
 */
 export interface QueryResultDtoOfViaExport {
    /**
     * 
     * @type {number}
     * @memberof QueryResultDtoOfViaExport
     */
    totalItems?: number;
    /**
     * 
     * @type {Array<ViaExport>}
     * @memberof QueryResultDtoOfViaExport
     */
    items?: Array<ViaExport>;
    /**
     * 
     * @type {CustomGridRenderOfViaExport}
     * @memberof QueryResultDtoOfViaExport
     */
    gridRender?: CustomGridRenderOfViaExport;
}

/**
 * 
 * @export
 * @interface QueryResultDtoVai
 */
 export interface QueryResultDtoVai extends QueryResultDtoOfViaExport {
    /**
     * 
     * @type {number}
     * @memberof QueryResultDtoVai
     */
    skipAmount?: number;
    /**
     * 
     * @type {Array<SkipManager>}
     * @memberof QueryResultDtoVai
     */
    skipManager?: Array<SkipManager>;
}

export interface ViaExportHardwareGrid {
	ViaExportHardwareGridResult: QueryResultDtoVai | null;
	filter: FilterValueDto[] | null;
}
export interface ViaExportSoftwareGrid {
	ViaExportSoftwareGridResult: QueryResultDtoOfViaExport | null;
	filter: FilterValueDto[] | null;
}

export const GET_GRID_VIA_EXPORT_HARDWARE = "GET_GRID_VIA_EXPORT_HARDWARE";
export const GET_FILTER_VIA_EXPORT_HARDWARE = "GET_FILTER_VIA_EXPORT_HARDWARE";

export const GET_GRID_VIA_EXPORT_SOFTWARE = "GET_GRID_VIA_EXPORT_SOFTWARE";
export const GET_FILTER_VIA_EXPORT_SOFTWARE = "GET_FILTER_VIA_EXPORT_SOFTWARE";